DVCSMP-690 Aeon multisensor (original) updates

greens opened the pull request:
@mckeed : Not sure the exact way you guys do pull requests, but I figure this can be a learning experience.

greens committed:
DVCSMP-690 Aeon multisensor improvements

* Removes the configure tile.
* Configuration commands should now be sent on wake-up, until a temperature report is sent
* Added preferences to configure the motion timeout and the reporting interval. Defaults are 20s and 5min
* Send an association remove during configuration to prevent double reports

greens committed:
DVCSMP-690 Aeon multisensor improvements

Undoing autoformatting.

mckeed commented on devicetypes/smartthings/aeon-multisensor.src/aeon-multisensor.groovy:
```diff
@@ -107,20 +110,30 @@ def updated(){
+	// try to configure until we've received a temperature update
+	if (!device.currentValue("temperature")) {
+		log.debug "temp is null, configuring"
+		results += configure()
```
You'll also need to send at least some of configure() here in the case that preferences (timeout or period) have been changed since the last wakeup. See aeon-multisensor-6 for an example, where it does `setConfigured("false")` in `updated()`. `configure()` is only automatically called when the device joins, `updated()` is called whenever the preferences are changed.

greens replied:
Ah, okay. I wasn't sure whether/where there was some function that was fired on preference update or not.

mckeed commented on devicetypes/smartthings/aeon-multisensor.src/aeon-multisensor.groovy:
```diff
@@ -85,12 +86,14 @@ metadata {
+
+	preferences {
+		input name: "timeout", type: "number", title: "Motion timeout", description: "Enter guard interval before receiving another motion event, in seconds", range: "20..60", defaultValue: 20
+		input name: "period", type: "number", title: "Reporting period", description: "Enter reporting period, in minutes", range: "4..15", defaultValue: 5
```
You should mention the range in the description text, like "Enter guard interval before receiving another motion event, in seconds (20-60)"

greens committed:
DVCSMP-1524 Remove polling capability from specifically excluded devices

greens committed:
DVCSMP-2220 Aeon Siren not reporting secure inclusion failure

Updated the parse logic to match other secure ZWave devices.

greens committed:
DVCSMP-690 Aeon Multisensor updates

Removes the battery from the sensor reporting group. It now reports every 12 hours

Renamed/reworked preferences to more closely match aeon-multisensor-6

Configuration will now happen on preference update or wakeup if it hasn't happened yet

greens commented:
@mckeed Didn't mean to have all of these other commits in this pull request as well. I'll remember in the future to do separate issue work on branches. I can go do the revert runaround, but they're here in the meantime if you want to take a look.

greens commented on devicetypes/smartthings/aeon-multisensor.src/aeon-multisensor.groovy:
```diff
@@ -31,6 +31,7 @@ metadata {
 		status "no motion (basic)"  : "command: 2001, payload: 00"
 		status "motion (binary)"    : "command: 3003, payload: FF"
 		status "no motion (binary)" : "command: 3003, payload: 00"
+		status "wakeup" 			: "command: 8407, payload: "
```
Just thought I'd point out that this was never showing up for me when I tested on the simulator so I would do a raw command of `8407`, but that was giving me outofboundsexceptions so I went with `840700` and that seemed to do what I wanted.

marstorp replied:
Although simulator is still somewhat used today it is not up to date with the current platform which may explain the issues you are seeing.

greens committed:
Revert "DVCSMP-2220 Aeon Siren not reporting secure inclusion failure"

This reverts commit cdd0beaafc654becb694e5370ede551e3f499071.

greens committed:
Revert "DVCSMP-1524 Remove polling capability from specifically excluded devices"

This reverts commit 97f88766715be7aca7059bb01c9125734f0832e1.

dsainteclaire commented on devicetypes/smartthings/aeon-multisensor.src/aeon-multisensor.groovy:
```diff
@@ -85,12 +86,18 @@ metadata {
+		details(["motion", "temperature", "humidity", "illuminance", "battery"])
+	}
+
+	preferences {
```
Because this will be visible to the user when they install it, I wonder if there's a better way to explain it to the user?  Thoughts @marstorp

greens replied:
`		input description: "Please consult AEOTEC MULTISENSOR 6 operating manual for advanced setting options. You can skip this configuration to use default settings",
				title: "Advanced Configuration", displayDuringSetup: true, type: "paragraph", element: "paragraph"`

This explanation makes an appearance in the multisensor 6 DTH that I cribbed this from.

marstorp replied:
Sounds like a reasonable description to me :)
However with default value set to "${motionDelayTime}" where motionDelayTime isn't defined can the user really skip this step?

greens replied:
The code in configure where the variables are used checks to see if they exist first and uses defaults if not.

marstorp replied:
yes, true but when you enter the preference page can you skip it or do you have to enter a value?

greens replied:
Just checked on my site. It looks like the field is blank. If I want the defaults to be 20s and 5 minutes, should I use 

`defaultValue: "20 seconds"`

or 

`defaultValue: 0`

greens replied:
From the [docs](http://docs.smartthings.com/en/latest/device-type-developers-guide/device-preferences.html?highlight=defaultValue):

> Setting a default value (defaultValue: "foobar") for an input may render that selection in the mobile app, but the user still needs to enter data in that field. It’s recommended to not use defaultValue to avoid confusion.

So the defaultValue field seems to be useless?

dkirker replied:
Yeah... defaultValue is not really used properly (I need to see if there is a mobile ticket open for this). It really gets bad in Android if required:true is set. I think it might work in iOS (it did for me). Check out PROB-1673. If you look at the Aeon Siren code, it has a good use of enforcing defaultValue configuration and making sure that updated values are sent to the device.

dsainteclaire commented:
Just the one comment about the preferences change.  Otherwise I don't see anything glaring.  Good job!

marstorp commented on devicetypes/smartthings/aeon-multisensor.src/aeon-multisensor.groovy:
```diff
@@ -102,25 +109,39 @@ def installed(){
 def updated(){
```
just want to point out an issue with our platform, sometimes the devices are added silently, i.e. the updated() method is not called from the mobile app.
If this is required to be executed on install then this will have to be triggered from the install method too :(

greens replied:
I think as written it's only expected to be called on a preference change.

munds replied:
This was an issue I saw in other DTHs. Are we retroactively adding that as "best practice" now? @marstorp

marstorp replied:
if it's only expected to be called on preference change I'm good with this, although I wonder if the user is presented with the choice to change the presences at install if it's a silent install.
But I'm fine with this as is :)

dkirker replied:
@marstorp Shouldn't matter either way. Now the logic is: configure() should get called by the platform (which is perfect for when the device is powered via the USB port), this device will also receive the configure() during the Z-Wave WakeUpNotification if the device is battery powered.

greens committed:
DVCSMP-690 Aeon Multisensor updates

Added some extra description text to the configuration preferences.

workingmonk commented:
can you add a short description to the PR title after the ticket? makes it easier for the deploy blog
